src/main/java/org/gridsuite/directory/server/DirectoryController.java

```diff
@@ -37,104 +37,98 @@ public DirectoryController(DirectoryService service) {
     })
     public ResponseEntity<Mono<Void>> setAccessRights(@PathVariable("elementUuid") UUID elementUuid,
-                                                      @RequestBody boolean isPrivate,
+                                                      @RequestParam("private") boolean isPrivate,
                                                       @RequestHeader("userId") String userId) {
         return ResponseEntity.ok().body(service.setAccessRights(elementUuid, isPrivate, userId));
     }
 
-    @DeleteMapping(value = "/directories/{elementUuid}")
+    @DeleteMapping(value = "/elements/{elementUuid}")
     @Operation(summary = "Remove directory/element")
     @ApiResponses(@ApiResponse(responseCode = "200", description = "Directory/element was successfully removed"))
     public ResponseEntity<Mono<Void>> deleteElement(@PathVariable("elementUuid") UUID elementUuid,
                                                     @RequestHeader("userId") String userId) {
         return ResponseEntity.ok().body(service.deleteElement(elementUuid, userId));
     }
 
-    @RequestMapping(method = RequestMethod.HEAD, value = "/directories/{directoryUuid}/{elementName}")
-    @Operation(summary = "Check if the element exists")
+    @RequestMapping(method = RequestMethod.HEAD, value = "/elements/{elementUuid}/elements/{elementName}")
+    @Operation(summary = "Check if a sub element exists")
     @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The element exists"),
-                           @ApiResponse(responseCode = "404", description = "The element doesn't exist")})
-    public ResponseEntity<Mono<Void>> elementExists(@PathVariable("directoryUuid") UUID directoryUuid,
+        @ApiResponse(responseCode = "404", description = "The element doesn't exist")})
+    public ResponseEntity<Mono<Void>> elementExists(@PathVariable("elementUuid") UUID elementUuid,
                                                     @PathVariable("elementName") String elementName,
                                                     @RequestHeader("userId") String userId) {
-        return ResponseEntity.ok().body(service.elementExists(directoryUuid, elementName, userId));
+        return ResponseEntity.ok().body(service.elementExists(elementUuid, elementName, userId));
     }
 
-    @GetMapping(value = "/directories/elements")
-    @Operation(summary = "get element infos from ids given as parameters")
-    @ApiResponses(value = {@ApiResponse(responseCode = "200", description = "The elements information")})
-    public ResponseEntity<Flux<ElementAttributes>> getElementsAttributes(@RequestParam("id") List<UUID> ids) {
-        return ResponseEntity.ok().body(service.getElementsAttribute(ids));
-    }
-
-    @PutMapping(value = "/directories/{elementUuid}/notify-parent")
-    @Operation(summary = "notify parent")
+    @PostMapping(value = "/elements/{elementUuid}/notification")
```

Comment:
/elements/{elementUuid}?notify=true ?

Reply:
Done

Reply:
if notify == false ?

Reply:
nothing append?
This is not the cleanest part of our API but maybe it is still better that /elements/{elementUuid}/notification" ?

Reply:
ok